### ISSUE TITLE
ComputerCraft based crafting provider

### ISSUE THREAD

**Issue by ruifung** [RS485/LogisticsPipes#117]
The idea is like this:
- A computer registers a item it can craft with a crafting pipe, and its required materials.
  - Computers can register multiple items it can craft.
- When a crafting request for one of the items is issued, fire a crafting event on the computer.
- Then wait for the result item to be in the attached inventory and extract it.

This would allow for programmable crafting lines, using CC based crafting to craft items.
Basically all it does is send the crafting event to the computer, and the computer handles requesting items from network and etc.

**Comment by Flow86** [comment]
+1 I like the idea

**Comment by davboecki** [comment]
The biggest problem with CC is that all it's work is threaded. CC accessing LP isn't that difficult because you can let the CC thread wait for the Main MC thread. But letting the Main thread wait for an CC thread will only cause a lot of mess and of course lag. So the first difficult thing to figure out for this would be a general way of interaction in this direction.

**Comment by ruifung** [comment]
Similar to how LP currently interacts with cc, let the computer do everything and notify the pipe with functions. And the pipe notifies the computer with a event. Would that work?

**Comment by davboecki** [comment]
Like i sayed you would have to pause the main thread to wait for the computer and that is a no go.

**Comment by ruifung** [comment]
Oh.

**Comment by Flow86** [comment]
I thought thats the CC's eventqueue for? that you dont have to wait on the computer?

**Comment by davboecki** [comment]
Well you queue the event to the computer true but you would need to wait for an answer.

**Comment by Flow86** [comment]
why? if its specified like the one above it only fires the crafting event to the computer, and waits in the (peripheral-)inventory for the result. the current crafting pipe does wait for the result in a similar way isnt it? you only have to specify a "timeout" of the crafting request until the pipe waits for its completion. (perhaps configurable)

**Comment by AartBluestoke** [comment]
this is what threads are for - we could easly create a "notify CC" task, and stick in in the list of things that the worker threads do (primarily rebuild routing tables atm); stalling one of the 4 threads for a tick or 2 won't matter.

**Comment by ruifung** [comment]
Alternatively, have the computer notify the pipe that crafting is complete instead of checking for the crafting results.

**Comment by bziemons** [comment]
Needs reevaluation. With the new ItemSink CC module and the crafting modules in general, there is a framework to make a CC crafting module.

**Comment by bziemons** [comment]
Moved to https://github.com/RS485/LogisticsPipes/wiki/Feature-Requests
